src/test/java/riemann/java/client/tests/LocalhostResolveTest.java

```diff
@@ -0,0 +1,146 @@
+        setEnv(env);
+
+        LocalhostResolver.CustomEnvVarName = null;
+        LocalhostResolver.resetUpdateTimes();
+        Assert.assertEquals(0, LocalhostResolver.getLastUpdateTime());
+        Assert.assertEquals(0, LocalhostResolver.getLastNetUpdateTime());
+    }
+
+    @Test
+    public void testUpdateInterval() {
+        Assert.assertEquals(0, LocalhostResolver.getLastUpdateTime());
+        String hostname = LocalhostResolver.getResolvedHostname();
+        long lastUpdateTime = LocalhostResolver.getLastUpdateTime();
+        Assert.assertNotNull(hostname);
+        try {
+            Thread.sleep(1500);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+
+        Assert.assertNotSame(lastUpdateTime, LocalhostResolver.getLastUpdateTime());
+    }
+
+    @Test
+    public void testNoEnvVars() throws UnknownHostException {
+        env.remove(LocalhostResolver.HOSTNAME);
+        env.remove(LocalhostResolver.COMPUTERNAME);
+        setEnv(env);
+
+        String hostname = LocalhostResolver.getResolvedHostname();
+        Assert.assertNotNull(hostname);
+
+        try {
+            Thread.sleep(1500);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+
+        // ensure queried hostname without env vars
+        Assert.assertEquals(LocalhostResolver.getLastUpdateTime(), LocalhostResolver.getLastNetUpdateTime());
+        Assert.assertEquals(java.net.InetAddress.getLocalHost().getHostName(), hostname);
+    }
+
+    @Test
+    public void testEnvVarWindows() {
+        System.getProperties().put(OS_NAME_PROP, "Windows7");
+
+        String hostname = LocalhostResolver.getResolvedHostname();
+        Assert.assertEquals(ExpectedWinHostname, hostname);
+        Assert.assertEquals(0, LocalhostResolver.getLastNetUpdateTime());
+    }
+
+    @Test
+    public void testEnvVarNix() {
+        System.getProperties().put(OS_NAME_PROP, "Linux");
+        String hostname = LocalhostResolver.getResolvedHostname();
+        Assert.assertEquals(ExpectedNixHostname, hostname);
+        Assert.assertEquals(0, LocalhostResolver.getLastNetUpdateTime());
+    }
+
+    @Test
+    public void testCustomEnvVar() {
+        final String customHostnameEnvVar = "AWS_HOST";
+        final String customHostname = "EC2-LocalHostResolverTest";
+        env.put(customHostnameEnvVar, customHostname);
+        setEnv(env);
+
+        LocalhostResolver.CustomEnvVarName = customHostnameEnvVar;
+        String hostname = LocalhostResolver.getResolvedHostname();
+        Assert.assertEquals(customHostname, hostname);
+        Assert.assertEquals(0, LocalhostResolver.getLastNetUpdateTime());
+    }
+
+
+    /**
+     * evil hack for testing (only!) with env var in-memory modification
+     * see: http://stackoverflow.com/a/7201825/1469004
+     *
+     * @param newEnv - to set in memory
+     */
+    protected static void setEnv(Map<String, String> newEnv) {
+        try {
```

Comment:
why??? Just add setEnv() in the production code and document it's a testing hook (which overrides getEnv()).